MatrixSDK/MXRestClient.m

```diff
@@ -3673,6 +3673,32 @@ - (MXHTTPOperation*) uploadContent:(NSData *)data
                                  }];
 }
 
+- (MXHTTPOperation*) maxUploadSize:(void (^)(NSInteger maxUploadSize))success
+                           failure:(void (^)(NSError *error))failure
+{
+    MXWeakify(self);
+    return [httpClient requestWithMethod:@"GET"
+                                    path:[NSString stringWithFormat:@"%@/config", contentPathPrefix]
+                              parameters:nil
+                                 success:^(NSDictionary *JSONResponse) {
+                                     MXStrongifyAndReturnIfNil(self);
+
+                                     if (success)
+                                     {
+                                         __block NSInteger maxUploadSize;
+                                         [self dispatchProcessing:^{
+                                             MXJSONModelSetInteger(maxUploadSize, JSONResponse[@"m.upload.size"]);
```

Comment:
The spec [mentions](https://matrix.org/docs/spec/client_server/r0.6.1#id406) that `m.upload.size` can be missing or null. Does this need to be handled here?

Reply:
`MXJSONModelSetInteger` does the check for you. We could default to -1 if the server does not provide it

Reply:
Done and `MXTools` now checks for a target size > 0 before assigning the limit.